### PR TITLE
Add good/bad angle information to analysis.json

### LINKED ISSUE #2130: Add good/bad particle info to analysis json output

### PR DESCRIPTION
**Pull request by akenmorris**
Resolves #2130 

* #2130 

This PR adds the cosine apex angle for each correspondence point in the particle system.  This value can then be thresholded at runtime the way Studio currently does to show particles as green or red (good or bad).

The structure is an array of arrays.  One array per anatomy/domain.

    "good_bad_angles": [
        [
            0.9853368372884771,
            0.9973483833053962,
             ....
        ],
        [
          .....
        ]
       ]

### CODE REVIEW

File: Libs/Particles/ParticleNormalEvaluation.h [hunk 1]
```diff
@@ -16,8 +16,10 @@ namespace shapeworks {
 class VtkMeshWrapper;
 class ParticleNormalEvaluation {
  public:
-  static std::vector<bool> evaluate_particle_normals(const Eigen::MatrixXd& particles, const Eigen::MatrixXd& normals,
-                                                     double max_angle_degrees);
+  static std::vector<double> evaluate_particle_normals(const Eigen::MatrixXd& particles,
+                                                       const Eigen::MatrixXd& normalss);
```
Comment (JakeWags) [inline Libs/Particles/ParticleNormalEvaluation.h:20]: Is this meant to be `normalss`?

File: Testing/data/shapeworksTests/analyze/analysis_baseline.json [hunk 2]
```diff
@@ -1285,5 +1285,73 @@
             "sample_3_0.vtk"
         ]
     ],
-    "groups": []
+    "groups": [],
+    "good_bad_angles": [
```
Comment (JakeWags) [inline Testing/data/shapeworksTests/analyze/analysis_baseline.json:1289]: In each domain, is the good-bad angles list in the same order as the normal particle list?

**Review by JakeWags — APPROVED** [review]
One small typo (I think), and some clarification needed. Otherwise, looks great

File: Libs/Particles/ParticleNormalEvaluation.h (hunk 1 above)
Comment (akenmorris) [reply Libs/Particles/ParticleNormalEvaluation.h]: Typo, yep.

File: Testing/data/shapeworksTests/analyze/analysis_baseline.json (hunk 2 above)
Comment (akenmorris) [reply Testing/data/shapeworksTests/analyze/analysis_baseline.json:1289]: Yes, same order as the particles in each particle file.

**Pull request merged**